File: data/derived_cot/rq1_traces/translation/r1/java_to_cpp/code_output/Thermostat.cpp
#include <string>

class Thermostat {
public:
    double currentTemperature;
    
    Thermostat(double currentTemperature, double targetTemperature, std::string mode) 
        : currentTemperature(currentTemperature), 
          targetTemperature(targetTemperature), 
          mode(mode) {}

    double getTargetTemperature() const {
        return targetTemperature;
    }

    void setTargetTemperature(double temperature) {
        targetTemperature = temperature;
    }

    std::string getMode() const {
        return mode;
    }

    bool setMode(std::string m) {
        if (m == "heat" || m == "cool") {
            mode = m;
            return true;
        } else {
            return false;
        }
    }

    void autoSetMode() {
        if (currentTemperature < targetTemperature) {
            mode = "heat";
        } else {
            mode = "cool";
        }
    }

    bool autoCheckConflict() {
        if (currentTemperature > targetTemperature) {
            if (mode == "cool") {
                return true;
            } else {
                autoSetMode();
                return false;
            }
        } else {
            if (mode == "heat") {
                return true;
            } else {
                autoSetMode();
                return false;
            }
        }
    }

    int simulateOperation() {
        autoSetMode();
        int useTime = 0;
        if (mode == "heat") {
            while (currentTemperature < targetTemperature) {
                currentTemperature += 1;
                useTime++;
            }
        } else {
            while (currentTemperature > targetTemperature) {
                currentTemperature -= 1;
                useTime++;
            }
        }
        return useTime;
    }

private:
    double targetTemperature;
    std::string mode;
};

File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_py/code_output/Thermostat.py
class Thermostat:
    def __init__(self, current_temperature: float, target_temperature: float, mode: str):
        self.currentTemperature = current_temperature
        self._targetTemperature = target_temperature
        self._mode = mode

    def getTargetTemperature(self) -> float:
        return self._targetTemperature

    def setTargetTemperature(self, temperature: float) -> None:
        self._targetTemperature = temperature

    def getMode(self) -> str:
        return self._mode

    def setMode(self, mode: str) -> bool:
        if mode == "heat" or mode == "cool":
            self._mode = mode
            return True
        else:
            return False

    def autoSetMode(self) -> None:
        if self.currentTemperature < self._targetTemperature:
            self._mode = "heat"
        else:
            self._mode = "cool"

    def autoCheckConflict(self) -> bool:
        if self.currentTemperature > self._targetTemperature:
            if self._mode == "cool":
                return True
            else:
                self.autoSetMode()
                return False
        else:
            if self._mode == "heat":
                return True
            else:
                self.autoSetMode()
                return False

    def simulateOperation(self) -> int:
        self.autoSetMode()
        use_time = 0
        if self._mode == "heat":
            while self.currentTemperature < self._targetTemperature:
                self.currentTemperature += 1
                use_time += 1
        else:
            while self.currentTemperature > self._targetTemperature:
                self.currentTemperature -= 1
                use_time += 1
        return use_time